assignment2/commonAncestor.py

```diff
@@ -0,0 +1,48 @@
+"""Design an algorithm and write code to find the lowest common ancestor of two
+nodes in a binary tree. Avoid storing additional nodes in a data structure."""
+
+from binarytree import BinaryTree
+
+def commonAncestor(tree, key1, key2):
+    """Find the lowest common ancestors of the given nodes in the given Binary
+    Tree.
+
+    Args:
+        tree: a BinaryTree;
+        key1: an integer, key of the first node;
+        key2: an integer, key of the second node.
+
+    Return:
+        an integer, key of the lowest common ancestors of two given nodes,
+        or `None` if some of the given nodes doesn't exists in the given tree.
+    """
+    def _commonAncestor(cur):
```

Comment:
@flerdacodeu I don't want to change the approach, but I add more if-statments (that was covered or convoluted) and comments. Hope now it clearer.